src/constructs/lambda/lambda.ts

```diff
@@ -24,6 +25,14 @@ export interface GuFunctionProps extends GuDistributable, Omit<FunctionProps, "c
    * @see https://docs.aws.amazon.com/lambda/latest/dg/concurrent-executions.html
    */
   throttlingMonitoring?: GuLambdaThrottlingMonitoringProps;
+  /**
+   * SSM parameter for the S3 bucket that contains your lambda file.
+   *
+   * Typically you should not override this.
+   *
+   * @defaultValue  [[`GuDistributionBucketParameter`]]
+   */
+  bucketNamePath?: string;
```

Comment:
@akash1810 adding to aws-account-setup is indeed the plan. See:

https://github.com/guardian/cdk/pull/1789

I am unsure if we should special-case our dist bucket here in a conditional - it avoids the perils of teams going rogue as you say, but is also a very specific solution that would expand our interface but only benefit us. It seems likely we will want an escape hatch here for other purposes too. If we want to avoid that, it might be better for us to avoid the `GuLambdaFunction` construct in the Cognito work altogether and just use AWS's underlying stuff?(Which is okay just more verbose obviously.)

Reply:
> but is also a very specific solution that would expand our interface but only benefit us

👍🏽